src/stk/molecular/topology_graphs/host_guest/complex.py

```diff
@@ -133,36 +220,54 @@ def __init__(
             If `guest_start` or `guest_target` is defined but the other
             is not.
 
+        :class:`ValueError`
+            If the old API and new API are used simultaneously.
+
+        :class:`RuntimeError:
+            If no guests are provided.
+
         """
 
-        num_nones = sum(
-            1 for vector in (guest_start, guest_target)
-            if vector is None
-        )
-        if num_nones == 1:
-            raise TypeError(
-                'If guest_start or guest_target is defined, '
-                'the other must be too.'
+        if guests is not None and (
+            guest is not None or len({
+                displacement, guest_start, guest_target
+            }) > 1
+        ):
```

Comment:
Ok i've just realized why my previous suggestion would not work.


how about

```python
old_parameters = (guest, displacement, guest_start, guest_target)
if guests is not None and any(param is not None for param in old_parameters):

```

which I feel is better because you can read it and know what its doing and meant to do. The length stuff makes me a little uncomfrable because i think you have to think about it a little bit to understand what it is doing and why.

Reply:
I agree. I like it.